### DISCUSSION TITLE
how to simulate indentation using ALE

### DISCUSSION THREAD

**Discussion by dongzhangim** [OpenRadioss/OpenRadioss#2119] [Q&A]
Hello OpenRadioss Community, my first post here!
I want to know how to simulate the indentation process (2D or 3D) using the ALE option. I know that radioss has the ALE capablity from the taylor-rod simulation. However, in taylor rod simulation, no contact was assumed. 
I've checked the internet, and find that indentation using ALE was possible. https://community.altair.com/community/en?id=community_question&sys_id=22868c3a1b2bd0908017dc61ec4bcb34. 
But, the details are missing, such as 2d or 3d, what is the contact type (inter/type 9, 12 or 18)? Can the friction be considered?
Thanks and regards!
Dong

**Comment by MarianBulla** [comment]
Hi Dong,
first of all, you're very welcome in the OpenRadioss Community !  :-)
I'm very glad, you are interested in OpenRadioss.
Yes, you are right: Radioss has quite stropng ALE capabilities.
Regarding your question: Usually, for ALE, you don't need any cintact interface. As long as you stay only with ALE, everything works without contact interfaces. You need contact interfaces, if you are going for fluid-structure-interaction (FSI). Which means, you have ALE and Lagrangian parts.
But only for ALE, it works without interfaces. E.g.:
![image](https://github.com/OpenRadioss/OpenRadioss/assets/105412502/1123ad4f-8680-4f10-9f19-22a8030c2e14)
![image](https://github.com/OpenRadioss/OpenRadioss/assets/105412502/b913b546-fb44-42cd-9ea2-64f6aa98744d)


See example input-dec (without the geometry).

#RADIOSS STARTER
#---1----|----2----|----3----|----4----|----5----|----6----|----7----|----8----|----9----|---10----|
/BEGIN
ALE_Example                                                                     
       120         0
                   g                  cm                 mus
                   g                  cm                 mus
#---1----|----2----|----3----|----4----|----5----|----6----|----7----|----8----|----9----|---10----|
#-  1. CONTROL CARDS:
#---1----|----2----|----3----|----4----|----5----|----6----|----7----|----8----|----9----|---10----|
/TITLE

#---1----|----2----|----3----|----4----|----5----|----6----|----7----|----8----|----9----|---10----|
/ALE/DONEA
#              alpha               gamma             Fscalex             Fscaley             Fscalez
                   0                   0                   0                   0                   0
#               Vmin
                   0
#---1----|----2----|----3----|----4----|----5----|----6----|----7----|----8----|----9----|---10----|
#---1----|----2----|----3----|----4----|----5----|----6----|----7----|----8----|----9----|---10----|
#-  2. MATERIALS:
#---1----|----2----|----3----|----4----|----5----|----6----|----7----|----8----|----9----|---10----|
/MAT/BIMAT/1
impactor                                                                                        
               2.785                    
         4         5
                 0.0                 1.0
/ALE/MAT/1
                 1.0
#---1----|----2----|----3----|----4----|----5----|----6----|----7----|----8----|----9----|---10----|
/MAT/BIMAT/2
void                                                                                        
                   0                    
         4         5
                 0.0                 0.0
/ALE/MAT/2
                 1.0
#---1----|----2----|----3----|----4----|----5----|----6----|----7----|----8----|----9----|---10----|
/MAT/BIMAT/3
plate                                                                                        
               2.705                    
         4         5
                 1.0                 0.0
/ALE/MAT/3
                 1.0
#---1----|----2----|----3----|----4----|----5----|----6----|----7----|----8----|----9----|---10----|
/MAT/HYD_JCOOK/4
Platte
#              RHO_I               RHO_O
               2.705          
#                  E                  nu
               0.734                 .33
#                  a                   b                   n             eps_max           sigma_max
             0.00275             0.00255                0.30                   0                   0
#               Pmin                 Psh
               -5e-3                   0
#                  c               EPS_0                   m               Tmelt               Tmax
                                                           1                 792
#              RHOCP
            3.461e-5
/ALE/MAT/4
                 1.0
#---1----|----2----|----3----|----4----|----5----|----6----|----7----|----8----|----9----|---10----|
/EOS/GRUNEISEN/4
Platte
#                  C                  S1                
               0.533               1.338              
#                 GA                   A                  E0                Rho0
                2.18                 .48                   0               2.705 
#---1----|----2----|----3----|----4----|----5----|----6----|----7----|----8----|----9----|---10----|
/MAT/HYD_JCOOK/5
Impactor
#              RHO_I               RHO_O
               2.785          
#                  E                  nu
               0.734                 .33
#                  a                   b                   n             eps_max           sigma_max
             0.00265             0.00426                0.34                   0                   0
#               Pmin                 Psh
               -5e-3                   0
#                  c               EPS_0                   m               Tmelt               Tmax
               0.015                1e-6                   1                 775
#              RHOCP
            3.461e-5
/ALE/MAT/5
                 1.0
#---1----|----2----|----3----|----4----|----5----|----6----|----7----|----8----|----9----|---10----|
/EOS/GRUNEISEN/5
Impactor
#                  C                  S1                
               0.533               1.338              
                2.18                 .48                   0               2.785 
#---1----|----2----|----3----|----4----|----5----|----6----|----7----|----8----|----9----|---10----|
#-  3. NODES:
#---1----|----2----|----3----|----4----|----5----|----6----|----7----|----8----|----9----|---10----|
/NODE
         1                   0                   0                   0
.
.
.
    453838                   0          .407010417                   0
    453839                   0            .4169375                   0
    453840                   0          .426864583                   0
    453841                   0          .436791667                   0
    453842                   0           .44671875                   0
    453843                   0          .456645833                   0
    453844                   0          .466572917                   0
    453845                   0                   0        .00992708333
    453846                   0                   0         .0198541667
    453847                   0                   0           .02978125
    453848                   0                   0         .0397083333
#---1----|----2----|----3----|----4----|----5----|----6----|----7----|----8----|----9----|---10----|
#-  4. BOUNDARY CONDITIONS:
#---1----|----2----|----3----|----4----|----5----|----6----|----7----|----8----|----9----|---10----|
/BCS/1
New BCS 1
#  Tra rot   skew_ID  grnod_ID
   010 000         0         1
/GRNOD/NODE/1
BCS_group_1_of_NODE
         1        25        49        73        97       121       145       169       193       217
.
.
    452337    452338    452339    452340    452341    452342    453315
#---1----|----2----|----3----|----4----|----5----|----6----|----7----|----8----|----9----|---10----|
/BCS/2
New BCS 2
#  Tra rot   skew_ID  grnod_ID
   011 000         0         2
/GRNOD/NODE/2
BCS_group_2_of_NODE
    452391    452392    452393    452394    452395    452396    452397    452398    452399    452400
    452401    452402    452403    452404    452405
#---1----|----2----|----3----|----4----|----5----|----6----|----7----|----8----|----9----|---10----|
#-  5. PARTS:
#---1----|----2----|----3----|----4----|----5----|----6----|----7----|----8----|----9----|---10----|
/PART/1
PSHELL1
         1         1         0
#---1----|----2----|----3----|----4----|----5----|----6----|----7----|----8----|----9----|---10----|
# QUAD elements of /PART/1/PSHELL1:
#---1----|----2----|----3----|----4----|----5----|----6----|----7----|----8----|----9----|---10----|
/QUAD/1
         1         1         2        26        25
         2         2         3        27        26
         3         3         4        28        27
         4         4         5        29        28
         5         5         6        30        29
         6         6         7        31        30
         7         7         8        32        31
         8         8         9        33        32
         9         9        10        34        33
        10        10        11        35        34
        11        11        12        36        35
        12        12        13        37        36
        13        13        14        38        37
.
.
.
    451871    451027    453816    453817    451028
    451872    451028    453817    453818    452764
#---1----|----2----|----3----|----4----|----5----|----6----|----7----|----8----|----9----|---10----|
/PART/3
PSHELL3
         3         3         0
#---1----|----2----|----3----|----4----|----5----|----6----|----7----|----8----|----9----|---10----|
# QUAD elements of /PART/3/PSHELL3:
#---1----|----2----|----3----|----4----|----5----|----6----|----7----|----8----|----9----|---10----|
/QUAD/3
    228817    228395    228396    228420    228419
.
.
.
    279071    278587    279066    279067    278588
    279072    278588    279067    452405    452404
#---1----|----2----|----3----|----4----|----5----|----6----|----7----|----8----|----9----|---10----|
#-  6. GEOMETRICAL SETS:
#---1----|----2----|----3----|----4----|----5----|----6----|----7----|----8----|----9----|---10----|
/PROP/SOLID/1
solid
#   Isolid    Ismstr               Icpre               Inpts      Irot    Iframe                  dn
         0         0                   0                   0         0         0                   0
#                q_a                 q_b                   h            LAMBDA_V                MU_V
                   0                   0                   0                   0                   0
#             dt_min   istrain      IHKT
                   0         0         0
#---1----|----2----|----3----|----4----|----5----|----6----|----7----|----8----|----9----|---10----|
/PROP/SOLID/2
solid
#   Isolid    Ismstr               Icpre               Inpts      Irot    Iframe                  dn
         0         0                   0                   0         0         0                   0
#                q_a                 q_b                   h            LAMBDA_V                MU_V
                   0                   0                   0                   0                   0
#             dt_min   istrain      IHKT
                   0         0         0
#---1----|----2----|----3----|----4----|----5----|----6----|----7----|----8----|----9----|---10----|
/PROP/SOLID/3
solid
#   Isolid    Ismstr               Icpre               Inpts      Irot    Iframe                  dn
         0         0                   0                   0         0         0                   0
#                q_a                 q_b                   h            LAMBDA_V                MU_V
                   0                   0                   0                   0                   0
#             dt_min   istrain      IHKT
                   0         0         0
#---1----|----2----|----3----|----4----|----5----|----6----|----7----|----8----|----9----|---10----|
#-  7. INITIAL VELOCITIES:
#---1----|----2----|----3----|----4----|----5----|----6----|----7----|----8----|----9----|---10----|
/INIVEL/TRA/1
New INIVEL 1
#                 Vx                  Vy                  Vz   Gnod_id   Skew_id
                   0                   0                 -.7         3         0
/GRNOD/PART/3
INIVEL_group_3_of_PART
         1
#---1----|----2----|----3----|----4----|----5----|----6----|----7----|----8----|----9----|---10----|
/END
#---1----|----2----|----3----|----4----|----5----|----6----|----7----|----8----|----9----|---10----|


That's basically all.

Hope it helps you ?

Thanks and best regards,
Marian

**Reply by dongzhangim** [reply]
Dear Marian,
I see your point. The input file you shared maybe was to simulate the "fusion" of two different aluminum ( my guess) within an impact process. Because it behaves more like fluid, therefore, the contact (Coulomb friction) was not considered.

Regarding the ALE, I'm wondering how to control the mesh distortion in radioss. Sometimes, ALE in radioss will also have a mesh distortion problem. Like the Taylor-rod impact simulation https://github.com/orgs/OpenRadioss/discussions/1914#discussioncomment-7140695), when the equivalent strain is greater than 2.5~3, the simulation will stop due to the element excessive distortion. (FSI may solve this problem because the eulerian method will not distort meshes at all). This reminds me that Abaqus can control the frequency of remeshing in ALE. 

And it comes to my question. Are the basics on how to realize "ALE" different for Radioss and Abaqus? Abaqus simply does lagrangian simulation and then does remeshing and finally transfers data to new mesh?? it is actually not ALE? ?

Finally, for the indentation process,  I have succeeded in adding an elastic plane and contact with the rod with the help of inter/type 9 in the Taylor-rod simulation, which I guesss should be called FSI rather than ALE.
thanks and regards,
Dong

**Comment by Schwoertzig** [comment]
Hi @dongzhangim ,
**contact interfaces** are introduced to handle interaction between **two different meshes**.
For ALE in a single mesh you can handle automatically interaction between materials as long as you have a mixture law (by using multimat /MAT/LAW51, or multifluid /MAT/LAW151)

- /MAT/LAW51 will fit both subsonic & supersonic phenomena (staggered scheme)
- /MAT/LAW151 is recommended only for high speed phenomena (otherwise since it is a collocated scheme you may encounter the well known "low mach limitation" )

Regards
Thierry

**Reply by dongzhangim** [reply]
Hi Thierry, 
For the multimat (law51 and 151) you said, I would check the reference guide and the example "NACA2929J_30fps_v3_0000.rad". 
I have 3 other questions for the CEL(coupled Eulerian and Lagrangian similar to Abaqus) which uses multimat.
1 Are the contact options similar to the ALE? type 1, 9, 12, 18, and 22 (from the reference guide of radioss 2022)? 
2 For the EOS parameters which also have to be set when the IFORM > 0 (allows Johnson-Cook model for metal) for law51, the P0 (initial total pressure) set to 0.1 MPa or 0 is OK for solid metals? Are there any examples available for the CEL simulation for metals (Solids for both Lagrangian part and the Eulerian part with contacts as well)
3 can the ALE model Marian shared be transformed into a CEL model? The plane is set to be lagrangian and the impactor stays as Eulerian. Law20/Bimat for 2D analysis and Inter/type 9 can also be used?? EOS for bimat can be deleted?@MarianBulla @Schwoertzig 
thanks and regards,
Dong
